lib/tree.rb

```diff
@@ -1,38 +1,69 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
-
-  def initialize
+class Tree
+  attr_reader :height, :age, :apples, :alive
+
+  def initialize(height = 0, age = 0, apples = [], alive = true)
+    @height = height
+    @age = age
+    @apples = apples
+    @alive = alive
   end
 
   def age!
+    # Age our tree by one year
+    @age += 1
+    
+    # Grow our tree a rand height between 1-3 feet per year
+    @height += rand(1..3)
+    
+    # Wait for our tree to age until 5, grow some fruit, then steal it
+    if @age >= 5
+      rand(100..300).times{ add_apples }
+    end
+  
+    # Our poor tree dies exactly on its 30th birthday, assuming its fruit has been picked
+    if @age >= 30
+      @alive = false
+    end
   end
 
   def add_apples
+    # Push a new red apple. Choose a random diameter size between 5-10 inch 
+    @apples.push Apple.new('red', rand(4..10))
   end
 
   def any_apples?
+    @apples.count > 0
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.pop
   end
 
   def dead?
+    !@alive
   end
 end
 
 class Fruit
+  attr_reader :has_seeds
+
   def initialize
-    has_seeds = true
+    @has_seeds = true
   end
-end
+end 
```

Comment:
Got some trailing white space here. If you're a vim user, I would add `set list` to your vimrc file so it highlights unwanted white space.

[Here's some documentation about it.](http://vim.wikia.com/wiki/Highlight_unwanted_spaces)